dvc/fs/http.py

```diff
@@ -81,11 +81,17 @@ def _prepare_credentials(self, **config):
                     f"Auth method {auth_method!r} is not supported."
                 )
 
-        if "ssl_verify" in config:
-            with fsspec_loop():
+        with fsspec_loop():
+            if "ssl_verify" in config:
                 client_kwargs["connector"] = aiohttp.TCPConnector(
-                    ssl=make_context(config["ssl_verify"])
+                    enable_cleanup_closed=True,
+                    ssl=make_context(config["ssl_verify"]),
                 )
+            else:
+                client_kwargs["connector"] = aiohttp.TCPConnector(
+                    enable_cleanup_closed=True
+                )
```

Comment:
Is there more detail on this? I'd like to see why this is not the default in `aiohttp`.

Reply:
Looks like aiohttp used to do this by default 5 years back but was changed by this commit https://github.com/aio-libs/aiohttp/commit/02b0951b1739048b8a8b80a2b406e620e5e17dce as an option. But I could not find any reasoning.

Reply:
From https://docs.aiohttp.org/en/stable/client_reference.html#tcpconnector:

> Some ssl servers do not properly complete SSL shutdown process, in that case asyncio leaks SSL connections. If this parameter is set to True, aiohttp additionally aborts underlining transport after 2 seconds. It is off by default.

From the description of how it works, it seems more reasonable that this isn't the default.

Reply:
I was too wondering why this is not the default, but digging around could not find any reasoning either. I guess this only becomes an issue under particular network conditions (broken TLS connection?), which were often encountered in dvc bench due to our large number of concurrent requests in dvc bench.

Reply:
Somehow relevant https://github.com/aio-libs/aiohttp/pull/5406

Reply:
It looks like this could be the context for the change: https://github.com/aio-libs/aiohttp/issues/1767.

Reply:
I guess we could also ask 😄